dbms/src/Common/Arena.h

```diff
@@ -93,6 +97,11 @@ class Arena : private boost::noncopyable
     /// Add next contiguous chunk of memory with size not less than specified.
     void NO_INLINE addChunk(size_t min_size)
     {
+        if (resize_callback != nullptr)
+        {
+            if (!resize_callback())
+                throw ResizeException("Error in arena resize");
+        }
```

Comment:
Will there be a performance regression?

Reply:
Add unlikely for exception throw path? And add failpoint here?

Reply:
OK, I will add unlikely for it, but for failpoint, actually it is not a standalone behavior, because if this function throws error, the related partition must be marked as to spill, otherwise, it may cause wrong result. I will add failpoint to test this case in test pr.